libmcount/mcount.c

```diff
@@ -1360,6 +1364,8 @@ static unsigned long __mcount_exit(long *retval)
 
 	rstack = &mtdp->rstack[mtdp->idx - 1];
 
+	sym = find_symtabs(&mcount_sym_info, rstack->child_ip);
+	rstack->size = sym->size;
```

Comment:
This will add more overhead in the fast path.  I think it should call the function only if the size filter is specified.